### ISSUE TITLE
How to decorate each instance in an ActiveRecord collection and infer the decorator class from an attr on each instance?

### ISSUE THREAD

**Issue by rahilsondhi** [drapergem/draper#678]
I have a class that has a `type` attr. I defined `#decorator_class` which does `"#{type.camelize}CarDecorator".constantize`. This means if `car = Car.last` has a `type` of `red`, then doing `car.decorate` will use the `RedCarDecorator` - cool!

Now, I want to decorate a whole collection, like `Car.all.decorate` and I want the decorator class to be inferred by each instances' `type`. That doesn't happen because the scope chain `decorate` method is a class method and doesn't have access to the instances' `type` attr: https://github.com/drapergem/draper/blob/master/lib/draper/decoratable.rb#L52-L61

tldr: How do I decorate each instance in an ActiveRecord collection and infer the decorator class **based on an attr of each instance**?

**Comment by lucianosousa** [comment]
@rahilsondhi the .decorate_collection method works like a charm for this case
https://github.com/drapergem/draper/blob/master/spec/draper/decorator_spec.rb#L117
